src/OrchardCore/OrchardCore.Media.Abstractions/Events/MediaCreationContext.cs

```diff
@@ -0,0 +1,15 @@
+using System.IO;
+
+namespace OrchardCore.Media.Events
+{
+    /// <summary>
+    /// The media context for the event of creating file.
+    /// </summary>
+    public class MediaCreationContext : MediaContext
+    {
```

Comment:
`MediaCreatingContext`

Also we need to figure out how the stream dispose will be handled. I need to think about the best way for that. Usings are tricky on disposable objects, when they get wrapped into a context.

Have you actually got a use case for this that you are implementing? 
If so provide code please, so I can test it, particularly dispose if the input stream is altered.

Think might want a unit test for this, will have to think some more.

Reply:
When top level caller opens stream then Stream object should be passed as explicit method parameter and not be wrapped by Context class. Passing it as method argument is kind of indication that stream should not leave call stack context.